device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttDeviceMethod.java

```diff
@@ -105,18 +99,6 @@ public void send(final IotHubTransportMessage message) throws TransportException
                     throw new IllegalArgumentException("Request id cannot be null or empty");
                 }
 
-                if (requestMap.containsKey(message.getRequestId()))
-                {
-                    if (requestMap.remove(message.getRequestId()) != DeviceOperations.DEVICE_OPERATION_METHOD_RECEIVE_REQUEST)
-                    {
-                        throwMethodsTransportException("Mismatched request and response operation");
-                    }
-                }
-                else
-                {
-                    throwMethodsTransportException("Sending a response for the method that was never invoked");
-                }
-
                 String topic = this.responseTopic + BACKSLASH +
```

Comment:
Note that neither this topic nor the message we send on the topic depends on anything saved in the map that is being deleted